src/core/capture.py

```
@@ -0,0 +1,190 @@
+            screenshot = sct.grab(monitor)
+            
+            # Convert to numpy array
+            img_array = np.array(screenshot)
+            
+            # mss returns BGRA, convert to RGB
+            rgb_array = img_array[:, :, [2, 1, 0]]  # BGR to RGB, drop alpha
+            
+            return rgb_array
+            
+    except Exception as e:
+        raise CaptureError(f"Failed to capture screen: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)